backend/tracim_backend/lib/translate/services/systran.py

```diff
@@ -67,8 +70,26 @@ def _translate_file(
         )
         if response.status_code == HTTPStatus.OK:
             return response.raw
+        elif response.status_code == HTTPStatus.FORBIDDEN:
+            raise TranslationServiceAccessRefused("access refused to systran translation service")
         else:
-            raise TranslationFailed(str(response.json()))
+            try:
+                error = response.json()["error"]
+            except Exception as exc:
+                raise TranslationServiceException(
+                    "Unknown error when requesting translation server"
+                ) from exc
+
+            if error.get("statusCode") == HTTPStatus.BAD_REQUEST:
+                raise InvalidParametersForTranslationService(
+                    "Invalid parameters given for translation: {}".format(error["message"])
+                )
```

Comment:
The 400 case happened in systran in case like giving a source or target language code that does not exist.